src/components/DeviceAction/rendering.js

```diff
@@ -0,0 +1,391 @@
+import Button from "../Button";
+import { NavigatorName } from "../../const";
+import Animation from "../Animation";
+import getDeviceAnimation from "./getDeviceAnimation";
+import GenericErrorView from "../GenericErrorView";
+import Circle from "../Circle";
+
+type RawProps = {
+  t: (key: string, options?: { [key: string]: string }) => string,
+};
+
+export function renderRequestQuitApp({
+  t,
+  device,
+}: {
+  ...RawProps,
+  device: Device,
+}) {
+  return (
+    <View style={styles.wrapper}>
+      <View style={styles.animationContainer}>
+        <Animation source={getDeviceAnimation({ device, key: "quitApp" })} />
+      </View>
+      <LText style={styles.text} semiBold>
+        {t("DeviceAction.quitApp")}
+      </LText>
+    </View>
+  );
+}
+
+export function renderRequiresAppInstallation({
+  t,
+  navigation,
+  appName,
+}: {
+  ...RawProps,
+  navigation: any,
+  appName: string,
+}) {
+  return (
+    <View style={styles.wrapper}>
+      <LText style={styles.text} semiBold>
+        {t("DeviceAction.appNotInstalled", { appName })}
+      </LText>
+      <View style={styles.actionContainer}>
+        <Button
+          event="DeviceActionRequiresAppInstallationOpenManager"
+          type="primary"
+          title={t("DeviceAction.button.openManager")}
+          onPress={() => navigation.navigate(NavigatorName.Manager)}
+          containerStyle={styles.button}
+        />
+      </View>
+    </View>
+  );
+}
+
+export function renderVerifyAddress({
+  t,
+  device,
+  currencyName,
+  onPress,
+  address,
+}: {
+  ...RawProps,
+  device: Device,
+  currencyName: string,
+  onPress?: () => void,
+  address?: string,
+}) {
+  return (
+    <View style={styles.wrapper}>
+      <View
+        style={[
+          styles.animationContainer,
+          device.modelId !== "blue" ? styles.verifyAddress : undefined,
+        ]}
+      >
+        <Animation source={getDeviceAnimation({ device, key: "validate" })} />
+      </View>
+      <LText style={[styles.text, styles.title]} semiBold>
+        {t("DeviceAction.verifyAddress.title")}
+      </LText>
+      <LText style={[styles.text, styles.description]}>
+        {t("DeviceAction.verifyAddress.description", { currencyName })}
+      </LText>
+      <View style={styles.actionContainer}>
+        {onPress && (
+          <Button
+            event="DeviceActionVerifyAddress"
+            type="primary"
+            title={t("common.continue")}
+            onPress={onPress}
+            containerStyle={styles.button}
+          />
+        )}
+        {address && (
+          <View>
+            <LText bold style={[styles.text, styles.title]}>
+              {address}
+            </LText>
+          </View>
+        )}
+      </View>
+    </View>
+  );
+}
+
+export function renderAllowManager({
+  t,
+  wording,
+  device,
+}: {
+  ...RawProps,
+  wording: string,
+  device: Device,
+}) {
+  // TODO: disable gesture, modal close, hide header buttons
```

Comment:
TODO comment still relevant?

Reply:
It depends on how we handle this on Luiz's manager branch